empress/tools.py

```diff
@@ -239,7 +238,7 @@ def match_inputs(
         )
 
     padded_metadata = sample_metadata.copy()
-    if len(sm_and_table_samples) < len(ff_table.columns):
+    if len(sm_and_table_samples) < len(ff_table.ids()):
```

Comment:
```suggestion
    if len(sm_and_table_samples) < len(table_samples):
```

Avoids recomputing `ff_table.ids()`. I guess this does technically assume that the sample IDs in the table are unique (otherwise the length of the set will be different from the length of `ff_table.ids()`), but I have to believe that that's already validated somewhere in biom...?

Reply:
Duplicate IDs are not allowed

Reply:
sick, now i can sleep better at night :D